rs/execution_environment/src/canister_manager.rs

```diff
@@ -2358,6 +2343,28 @@ impl CanisterManager {
     }
 }
 
+fn get_response_size(kind: &CanisterSnapshotDataKind) -> Result<u64, CanisterManagerError> {
+    if !matches!(&kind, CanisterSnapshotDataKind::WasmChunk { .. }) {
+        return Ok(CHUNK_SIZE);
+    }
+    let size = match kind {
+        CanisterSnapshotDataKind::WasmModule { size, .. } => *size,
+        CanisterSnapshotDataKind::MainMemory { size, .. } => *size,
+        CanisterSnapshotDataKind::StableMemory { size, .. } => *size,
+        CanisterSnapshotDataKind::WasmChunk { .. } => {
+            debug_unreachable!();
```

Comment:
Now that we don't have the "issue" with the artificial offset, we don't even the special case of `WasmChunk` (if `CHUNK_SIZE` was changed to be bigger than `MAX_SLICE_SIZE_BYTES`, then some test for actually uploading a chunk would fail anyway).